driver-core/src/main/com/mongodb/client/model/expressions/Expressions.java

```diff
@@ -84,9 +83,7 @@ public static DateExpression of(final Instant of) {
      * @return the string expression
      */
     public static StringExpression of(final String of) {
-        if (of == null) {
-            throw new IllegalArgumentException("string cannot be null");
-        }
+        Assertions.notNull("String", of);
```

Comment:
I used the type rather than the name, because `of` isn't very descriptive for the error message. I used `of` because it seems to prevent intellij label spam for a method that we are trying to make compact.

Reply:
> of isn't very descriptive for the error message

We try to use this check consistently (this is also how it was before my time). The usage in this PR is not consistent with how we tend to use the check, which is a price we may consider to pay if there is enough value gained as a result. What value does using `String` bring?

Reply:
"String cannot be null" is immediately clear, while "of cannot be null" sounds like a typo.